File: app/src/main/java/com/kelsos/mbrc/utilities/MediaButtonReceiver.java
package com.kelsos.mbrc.utilities;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import javax.inject.Inject;
import toothpick.Scope;
import toothpick.Toothpick;

public class MediaButtonReceiver extends BroadcastReceiver {
  @Inject MediaIntentHandler handler;
  private Scope scope;

  @Override
  public void onReceive(Context context, Intent intent) {
    if (scope == null) {
      scope = Toothpick.openScope(context.getApplicationContext());
      Toothpick.inject(this, scope);
    }

    handler.handleMediaIntent(intent);
  }
}

File: app/src/main/kotlin/com/kelsos/mbrc/utilities/MediaButtonReceiver.kt
package com.kelsos.mbrc.utilities

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import javax.inject.Inject
import roboguice.RoboGuice
import toothpick.Toothpick

class MediaButtonReceiver : BroadcastReceiver() {
  @Inject lateinit var handler: MediaIntentHandler

  override fun onReceive(context: Context, intent: Intent) {
    val scope = Toothpick.openScope(context.applicationContext)
    Toothpick.inject(this, scope)
    handler.handleMediaIntent(intent)
  }
}
